### DISCUSSION TITLE
table \"<>\" does not exist in source: <>"

### DISCUSSION THREAD

**Discussion by ajithv92** [hasura/graphql-engine#9269] [Question] [answered]
how to use metadata apis for create permissions (select, insert, delete etc) on Hasura. seems like using the template written in doc is not helping me..

all is get always is a BadRequest with 400 with error on body 

{
    "path": "$.args",
    "error": "table \"Devices\" does not exist in source: ISAXQA",
    "code": "not-exists"
}

![image](https://user-images.githubusercontent.com/5372258/205844893-e8bc3973-9fe5-4d54-9c3a-862c15c7e727.png)

![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/5372258/205845574-f8f67149-1ebf-45b4-987c-3a274ad682db.png)

would like some help on this !!

**Comment by ecthiender** [comment, marked as answer]
From the screenshot above, it looks like this is a MS SQL Server database. In that case, you'd have to use the SQL Server related metadata APIs (starts with `mssql_`). For creating select permission it would be `mssql_create_select_permission`. Full docs here: https://hasura.io/docs/latest/api-reference/metadata-api/permission/#mssql-create-select-permission

**Reply by ajithv92** [reply]
thank you !! also is there a way to do bulk updates like select, insert update, delete permission in one api request ?

**Reply by ecthiender** [reply]
Yes! Check out the [`bulk` query type](https://hasura.io/docs/latest/api-reference/metadata-api/index/#request-types) in the metadata API. You can add multiple metadata queries in one bulk query and make one API request.
